### ISSUE TITLE
subscribing to boolean variables

### ISSUE THREAD

**Issue by mrjono1** [ivantsov/redux-subscriber#5]
Hi ivantsov great project, you have the easiest to understand subscribing package for redux I could find.

I'm only getting started using this project but I have noticed when subscribing to Boolean properties the project does not seem to pickup the changes.

For my project I have just changed the Boolean properties to an integer with 1/0 values

Cheers
mrjono1

**Comment by ivantsov** [comment]
Hi @mrjono1, glad to hear that the project is useful for you :)

Regarding your problem - can you provide an example with the problem? I'm asking because I used the following code to reproduce the problem however it does work as expected:

```js
import {createStore} from 'redux';
import initSubscriber from 'redux-subscriber';

const initialState = {
  prop1: 123,
  prop2: true
};

function reducer(state = initialState, action) {
  if (action.type === 'CHANGE_STATE') {
    return {
      ...state,
      prop2: !state.prop2
    };
  }
  
  return state;
}

function action() {
  return {type: 'CHANGE_STATE'};
}

const store = createStore(reducer);
const subscribe = initSubscriber(store);

subscribe('prop2', state => console.log('state in subscriber', state));

console.log('initial state', store.getState());

store.dispatch(action());
store.dispatch(action());
store.dispatch(action());
```

**Comment by mrjono1** [comment]
Hi ivantosov 
so far I have been unable to create a small demo, I'm now thinking the issue I found was not a Boolean issue but something else. I have noticed issues when I changed nested properties.

object1.property1 = newvalue;
subscribe('object1.property1' - property always triggering

object1.object2.property1= newvalue;
subscribe('object1.object2.property1' - property not always triggering

Sorry for the late reply to my own submission.

I am new to redux so I could be doing something wong
